pkg/controlplane/instance.go

```diff
@@ -488,38 +464,18 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 	})
 
 	if utilfeature.DefaultFeatureGate.Enabled(apiserverfeatures.APIServerIdentity) {
-		m.GenericAPIServer.AddPostStartHookOrDie("start-kube-apiserver-identity-lease-controller", func(hookContext genericapiserver.PostStartHookContext) error {
-			kubeClient, err := kubernetes.NewForConfig(hookContext.LoopbackClientConfig)
-			if err != nil {
-				return err
-			}
-
-			leaseName := m.GenericAPIServer.APIServerID
-			holderIdentity := m.GenericAPIServer.APIServerID + "_" + string(uuid.NewUUID())
-
-			controller := lease.NewController(
-				clock.RealClock{},
-				kubeClient,
-				holderIdentity,
-				int32(IdentityLeaseDurationSeconds),
-				nil,
-				IdentityLeaseRenewIntervalPeriod,
-				leaseName,
-				metav1.NamespaceSystem,
-				labelAPIServerHeartbeat)
-			go controller.Run(hookContext.StopCh)
-			return nil
-		})
+		// the identity lease GC controller lives only in the Kube API server (it decides when to delete them) whereas
+		// the identity lease creating controller lives in k8s.io/apiserver because it is generally used by all API servers
 		m.GenericAPIServer.AddPostStartHookOrDie("start-kube-apiserver-identity-lease-garbage-collector", func(hookContext genericapiserver.PostStartHookContext) error {
```

Comment:
What's the reason for not moving the lease garbage collection too?

Reply:
The best way to avoid garbage being left around is to have the clean up logic be the same across all API servers and for it be handled by a cluster component like KAS or KCM.  Then we don't have to worry about what happens when an aggregated API server is removed.  Now if we decide to make the label values configurable in some way, then we would need to move this into a generic spot too.  Same goes for if we decide to allow a different namespace to be used.  This seems like the best place for this controller under the current code.